src/mono/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.WASI.xml

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<linker>
+	<assembly fullname="System.Private.CoreLib">
+		<!-- these methods are temporarily accessed via UnsafeAccessor from generated code until we have it in public API, probably in WASI preview3 and promises -->
```

Comment:
@lewing I removed it from the previous PR, but it was probably my mistake in testing.

If it's not there in `sample\wasi\http-p2` I get
```
System.AggregateException: One or more errors occurred. (Could not find RegisterWasiPollableHandle)
   ---> System.MissingMethodException: Could not find RegisterWasiPollableHandle
     at System.Net.Http.WasiHttpHandler.WasiEventLoop.RegisterWasiPollable(Pollable )
     at System.Net.Http.WasiHttpHandler.SendRequestAsync(OutgoingRequest )
     at System.Net.Http.WasiHttpHandler.SendAsync(HttpRequestMessage , CancellationToken )
     at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage , CancellationToken )
     at Test.Work()
```

I'm not clear why the `UnsafeAccessor`  in `System.Net.Http` doesn't keep it alive.

Reply:
If it's technically possible, could we please make any new test run on CI?

Reply:
hmm yeah that's weird, the trimmer does look at UnsafeAccessor: https://github.com/dotnet/runtime/blob/main/src/tools/illink/src/linker/Linker.Steps/UnsafeAccessorMarker.cs

maybe it's already trimmed out in the "library-linking" phase (where we run the trimmer on the shared framework in a special mode as part of the runtime build), can you check whether it's still there in the shared framework after the repo build?

Reply:
that doesn't apply to `DispatchWasiEventLoop` though since that one isn't called from the BCL itself so I'd expect it to be trimmed away by the library-linking phase.

Reply:
> maybe it's already trimmed out in the "library-linking" phase (where we run the trimmer on the shared framework in a special mode as part of the runtime build), can you check whether it's still there in the shared framework after the repo build?

None of `System.Private.CoreLib.dll` in my artifacts contain it, so it's trimmed pretty early.

> that doesn't apply to `DispatchWasiEventLoop` though since that one isn't called from the BCL itself so I'd expect it to be trimmed away by the library-linking phase.

`DispatchWasiEventLoop` is also trimmed.

There are many other things being protected in `src\mono\System.Private.CoreLib\src\ILLink\ILLink.Descriptors.xml` and `src\mono\System.Private.CoreLib\src\ILLink\ILLink.Descriptors.OSX.xml` for example.

I don't think I need to be fighting it too hard.

Reply:
The methods mentioned in those files are accessed from runtime native code so the trimmer can't possibly know about them, that shouldn't apply to `UnsafeAccessor`-marked methods, the trimmer should always preserve them.

I wonder if it's because the method is a local function _inside_ of another method.

Anyway, what you have is fine for now.

Reply:
It seems that helped!
I filled https://github.com/dotnet/runtime/issues/106009

Reply:
No, it failed again this time on CI [Log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-105879-merge-27339f9a976f41b4a7/Workloads-ST-Wasm.Build.Tests/1/console.82540b9a.log?helixlogtype=result)

I put the link descriptor back